File: cpp/643.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& n, int k) {
        int l = n.size(), s = 0;
        for (int i = 0; i < k; i++)
            s += n[i];
        int m = s;
        for (int i = k; i < l; i++) {
            s = s - n[i - k] + n[i];
            m = max(m, s);
        }
        return (m * 1.0) / k;
    }
};

File: python3/643.py
class Solution:
    def findMaxAverage(self, n: List[int], k: int) -> float:
        s = sum(n[:k])
        m = s
        for i in range(k, len(n)):
            s = s - n[i - k] + n[i]
            m = max(m, s)
        return m / k
